wgpu-native/src/instance.rs

```diff
@@ -301,11 +302,16 @@ pub extern "C" fn wgpu_create_surface_from_windows_hwnd(
     ))
 }
 
-pub fn request_adapter(
+pub type RequestAdapterCallback =
+    extern "C" fn(adapter: *const AdapterId, userdata: *mut c_void);
+
+pub fn request_adapter_async(
```

Comment:
do we have to have this internal method asynchronous at all? If we keep sync (basically, like `request_adapter` today), we can have the callback called once by any user of this method - that is simpler than trying to enforce that each return path is *separately* calling the function.

Reply:
Yeah we could do it either way. I think it's just a question of how much we want the implementations to vary (i.e. between the thin `pub extern "C" fn` vs. the inner `pub fn` that we're sharing with wgpu-remote), and how the event loop would be used in both.

Reply:
@kvark see updated commit – I just moved the synchronous parts into a separate function for now